apps/system-e2e/src/tests/judicial-system/regression/indictment-tests.spec.ts

```diff
@@ -0,0 +1,149 @@
+
+    // Case list
+    await page.goto('/krofur')
+    await page.getByRole('button', { name: 'Nýtt mál' }).click()
+    await page.getByRole('menuitem', { name: 'Ákæra' }).click()
+    await expect(page).toHaveURL('/akaera/ny')
+
+    // New indictment case
+    await page.getByTestId('policeCaseNumber0').click()
+    await page.getByTestId('policeCaseNumber0').fill(randomPoliceCaseNumber())
+    await page.getByText('Sakarefni *Veldu sakarefni').click()
+    await page.locator('#react-select-case-type-option-0').click()
+    await page.getByPlaceholder('Sláðu inn vettvang').click()
+    await page.getByPlaceholder('Sláðu inn vettvang').fill('Reykjavík')
+    await page.locator('input[id=arrestDate]').fill(today)
+    await page.keyboard.press('Escape')
+    await page
+      .getByRole('checkbox', { name: 'Ákærði er ekki með íslenska kennitölu' })
+      .check()
+    await page.getByTestId('nationalId').click()
+    await page.getByTestId('nationalId').fill('01.01.2000')
+    await page.getByTestId('accusedName').click()
+    await page.getByTestId('accusedName').fill(accusedName)
+    await page.getByTestId('accusedName').press('Tab')
+    await page.getByTestId('accusedAddress').fill('Testgata 12')
+    await page.locator('#defendantGender').click()
+    await page.locator('#react-select-defendantGender-option-0').click()
+    await Promise.all([
+      page.getByRole('button', { name: 'Stofna mál' }).click(),
+      verifyRequestCompletion(page, '/api/graphql', 'CreateCase').then(
+        (res) => (caseId = res.data.createCase.id),
+      ),
+      verifyRequestCompletion(page, '/api/graphql', 'Case'),
+    ])
+
+    // Police case files (Málsgögn)
+    await expect(page).toHaveURL(`/akaera/malsgogn/${caseId}`)
+    await Promise.all([
+      page.getByTestId('continueButton').click(),
+      verifyRequestCompletion(page, '/api/graphql', 'Case'),
+    ])
+
+    // Case file
+    await expect(page).toHaveURL(`/akaera/skjalaskra/${caseId}`)
+    await Promise.all([
+      page.getByTestId('continueButton').click(),
+      verifyRequestCompletion(page, '/api/graphql', 'Case'),
+    ])
+
+    // Processing
+    await Promise.all([
+      expect(page).toHaveURL(`/akaera/malsmedferd/${caseId}`),
+      verifyRequestCompletion(page, '/api/graphql', 'Case'),
+    ])
+    await Promise.all([
+      page.getByText('Játar sök').click(),
+      verifyRequestCompletion(page, '/api/graphql', 'UpdateDefendant'),
+    ])
+    await Promise.all([
+      page.getByTestId('continueButton').click(),
+      verifyRequestCompletion(page, '/api/graphql', 'Case'),
+    ])
+    // Case files
+    await expect(page).toHaveURL(`/akaera/domskjol/${caseId}`)
+
+    await uploadDocument(
+      page,
+      async () => {
+        await page
+          .locator('section')
+          .filter({
+            hasText:
+              'Ákæra *Dragðu gögn hingað til að hlaða uppVelja gögn til að hlaða upp',
+          })
+          .locator('button')
```

Comment:
Why not use `getByText` or `getByRole`?
```suggestion
          .getByText('Ákæra *Dragðu gögn hingað til að hlaða uppVelja gögn til að hlaða upp')
          .getByRole('button', {name: 'Ákæra *Dragðu gögn hingað til að hlaða uppVelja gögn til að hlaða upp'})
```